src/agent/market_schedule.rs

```diff
@@ -96,7 +96,7 @@ impl MarketSchedule {
         let month = when_local.date_naive().month0() + 1;
         let day = when_local.date_naive().day0() + 1;
         let time = when_local.time();
-        let weekday = when_local.weekday().number_from_monday().to_usize();
+        let weekday0 = when_local.weekday().number_from_monday().to_usize() - 1;
```

Comment:
I recommend adding a test for Monday too. `0-1` might overflow.

Reply:
Done